application/frontend/controllers/MfaController.php

```diff
@@ -161,6 +184,24 @@ public function actionVerify($mfaId)
             $value['clientExtensionResults'] = new stdClass();
         }
 
+        return $value;
+    }
+
+    /**
+     * @param $mfaId
+     * @return array|bool
+     * @throws HttpException
+     */
+    public function actionVerify($mfaId)
+    {
+        $messages = [
+            400 => \Yii::t('app', 'Mfa.InvalidCode'),
+            404 => \Yii::t('app', 'Mfa.VerifyFailure'),
+            429 => \Yii::t('app', 'Mfa.RateLimitFailure'),
+        ];
+
+        $value = self::getVerifyValue();
```

Comment:
Related to my comment above about returning `null` from the `getVerifyValue()` method:
```suggestion
        $value = self::getVerifyValue();
        if ($value === null) {
            throw new BadRequestHttpException(\Yii::t('app', 'Mfa.MissingValue'));
        }
```